src/components/atoms/IconInput/IconInput.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { ChangeEvent } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import "./IconInput.scss";
+
+export interface IconInputProps {
+  className?: string;
+  icon?: IconProp;
+  onChange?: (e?: ChangeEvent<HTMLInputElement>) => void;
+  value?: string;
+  autoComplete?: string;
+  placeholder?: string;
+  iconPosition?: "right" | "left";
+}
```

Comment:
We might want to split this up into `BaseIconInputProps` and then 2 variants (one for when there is an icon supplied, one for when no icon supplied) so that we can properly hint through the types that `iconPosition` is only going to be useful when `icon` is set, etc.

Reply:
Alternatively, and I think a better way to handle this would be to have 2 props:
`startIcon/endIcon`similar to what material-ui lib has: 
https://material-ui.com/components/text-fields/

Reply:
I'm ok with that. Avoids the potential runtime issues + is a cognitively simpler concept to grasp.

Reply:
@mike-lvov 
so `startIcon` / `endIcon` would be booleans and stand for position of the `icon`?
or
you suggest to replace `icon` prop by these 2 props and specify icon src there?

Reply:
They should accept both iconProp + rendered icon(to cover the case, when you want to add an onClick handler to an icon)

Reply:
For example this case should be covered: 
![image](https://user-images.githubusercontent.com/63194656/113858773-cb8b3880-97ac-11eb-8ff5-421f024f2d9a.png)

Reply:
so potentially, we can have both startIcon and endIcon for the input, right?

Reply:
@sunny-viktoryia yes. Plus, a developer should be able to pass an icon component from `@fortawesome/free-solid-svg-icons`. For ex. `faSearch` 

or he can pass a rendered Icon 
              ```
endIcon={<FontAwesomeIcon
                  className="chat-sidebar__controller__second-icon"
                  icon={faCommentDots}
                  size="lg"
                />}|
```

Reply:
added support for both side icons (`IconProp` or `JSX.Element`): https://github.com/sparkletown/sparkle/pull/1306/commits/4522f27ae790a9bc30e71c2e4e06433f7e2b9141